### PR TITLE
{2023.06}[foss/2023b] Extrae 4.2.0

### PR DESCRIPTION
**Pull request by ocaisa**
Alternative to #554

### CODE REVIEW

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
@ocaisa failing across the board?

**Comment by bedroge** [comment]
I checked the zen3 build:
```
checking for binutils... notfound
configure: libbfd library directory: /usr/lib/x86_64-linux-gnu
configure: Warning! Cannot find the libiberty library in the given binutils home. Please, make sure that the binutils packages is correctly i
nstalled. If you have installed the binutils package by hand from their source code, make sure that libiberty is installed. Some releases of
the binutils package do not install the libibery even invoking make install. The library should be within the libiberty directory within the
binutils source tree.
checking for bfd.h... no
configure: error: You have asked to gather call-site information through --with-unwind which must be translated using binutils, but either li
bbfd or libiberty are not found. Please make sure that the binutils-dev package is installed and specify where to find these libraries throug
h --with-binutils. The latest source can be downloaded from http://www.gnu.org/software/binutils
 (at easybuild/tools/run.py:682 in parse_cmd_output)
```

Probably we do still need the hook from #554?

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

**Comment by ocaisa** [comment]
Two different problems here, on Zen4 I see all the MPI tests failing. On ARM I see
```
Failed to isolate path for binutils libraries using /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/lib*/binutils/*-pc-linux-gnu/2.*
```

**Comment by ocaisa** [comment]
Easy fix for the second problem:
```
[ocaisa@login1 14044]$ ls /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/*/binutils/*-pc-linux-gnu/2.*
ls: cannot access '/cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/*/binutils/*-pc-linux-gnu/2.*': No such file or directory
[ocaisa@login1 14044]$ ls /cvmfs/software.eessi.io/versions/2023.06/compat/linux/aarch64/usr/*/binutils/*-linux-gnu/2.*
bfd-plugins  libbfd-2.40.0.gentoo-sys-devel-binutils-st.so  libctf-nobfd-2.40.0.gentoo-sys-devel-binutils-st.so  libiberty.a                                        libsframe.so
include      libbfd.so                                      libctf-nobfd.so                                      libopcodes-2.40.0.gentoo-sys-devel-binutils-st.so  libsframe.so.0
ldscripts    libctf-2.40.0.gentoo-sys-devel-binutils-st.so  libctf.so                                            libopcodes.so                                      libsframe.so.0.0.0
```

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by ocaisa** [comment]
All the Arm builds are failing 😭

**Comment by bedroge** [comment]
> All the Arm builds are failing 😭
> 
> ```
> FAIL: define_event_type_gen_pcf.sh
> FAIL: test-shared-library-nodebug.sh
> FAIL: test-shared-library-debug.sh
> FAIL: check_Extrae_user_function.sh
> ```

Weird, I also tested it on RISC-V (in the EESSI environment), and that worked fine :laughing:

**Comment by ocaisa** [comment]
I'll do a manual build tomorrow and gather the logs

**Comment by bedroge** [comment]
With the `bot/inspect.sh` script you can quickly access the logs. I found the following for `aarch64/generic`.

define_event_type_gen_pcf.sh.log
```
Extrae: Warning! API tries to initialize more than once
Extrae:          Previous initialization was done by API
./define_event_type_gen_pcf.sh: line 9: 136539 Segmentation fault      (core dumped) EXTRAE_ON=1 ./define_event_type_gen_pcf
merger: Output trace format is: Paraver
merger: Extrae 4.2.0
mpi2prv: Assigned nodes < aarch64-generic-node2.int.aws-rocky88-202310.eessi.io >
mpi2prv: Assigned size per processor < 0 bytes >
mpi2prv: File ./set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000136539000000000000.mpit is object 1.1.1 on node aarch64-generic-node2.int.aws-rocky88-202310.eessi.io assigned to processor 0
mpi2prv: Time synchronization has been turned off
fopen: No such file or directory
mpi2prv Error: Opening trace file ./set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000136539000000000000.mpit
AddFile_FS: Invalid argument
mpi2prv: Error creating file set
mpi2prv: Error! Some of the processors failed create trace descriptors
mpi2prv: An error has been encountered when generating the tracefile. Dying...
mpi2prv: Checking for target directory existence... exists, ok!
Error! Could not find 'Kernel execution'.
FAIL define_event_type_gen_pcf.sh (exit status: 1)
```

test-shared-library-nodebug.sh.log
```
./test-shared-library-nodebug.sh: line 7: 140950 Segmentation fault      (core dumped) EXTRAE_ON=1 ./main_nodebug
merger: Output trace format is: Paraver
merger: Extrae 4.2.0
mpi2prv: Assigned nodes < aarch64-generic-node2.int.aws-rocky88-202310.eessi.io >
mpi2prv: Assigned size per processor < 0 bytes >
mpi2prv: File ./set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000140950000000000000.mpit is object 1.1.1 on node aarch64-generic-node2.int.aws-rocky88-202310.eessi.io assigned to processor 0
mpi2prv: Time synchronization has been turned off
fopen: No such file or directory
mpi2prv Error: Opening trace file ./set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000140950000000000000.mpit
AddFile_FS: Invalid argument
mpi2prv: Error creating file set
mpi2prv: Error! Some of the processors failed create trace descriptors
mpi2prv: An error has been encountered when generating the tracefile. Dying...
mpi2prv: Checking for target directory existence... exists, ok!
Error! Could not find '0 (Unresolved)'.
FAIL test-shared-library-nodebug.sh (exit status: 1)
```

test-shared-library-debug.sh.log
```
./test-shared-library-debug.sh: line 7: 140957 Segmentation fault      (core dumped) EXTRAE_ON=1 ./main_debug
merger: Output trace format is: Paraver
merger: Extrae 4.2.0
mpi2prv: Assigned nodes < aarch64-generic-node2.int.aws-rocky88-202310.eessi.io >
mpi2prv: Assigned size per processor < 0 bytes >
mpi2prv: File ./set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000140957000000000000.mpit is object 1.1.1 on node aarch64-generic-node2.int.aws-rocky88-202310.eessi.io assigned to processor 0
mpi2prv: Time synchronization has been turned off
fopen: No such file or directory
mpi2prv Error: Opening trace file ./set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000140957000000000000.mpit
AddFile_FS: Invalid argument
mpi2prv: Error creating file set
mpi2prv: Error! Some of the processors failed create trace descriptors
mpi2prv: An error has been encountered when generating the tracefile. Dying...
mpi2prv: Checking for target directory existence... exists, ok!
Error! Could not find '0 (Unresolved)'.
FAIL test-shared-library-debug.sh (exit status: 1)
```

check_Extrae_user_function.sh.log
```
Welcome to Extrae 4.2.0
Extrae: Generating intermediate files for Paraver traces.
Extrae: Intermediate files will be stored in /tmp/bot/easybuild/build/Extrae/4.2.0/gompi-2023b/extrae-4.2.0/tests/functional/callstack
Extrae: Tracing buffer can hold 500000 events
Extrae: Tracing mode is set to: Detail.
Extrae: Successfully initiated with 1 tasks and 1 threads

begin = 0x401b80 v = (nil) end = 0x401bd4
Extrae: Intermediate raw trace file created : /tmp/bot/easybuild/build/Extrae/4.2.0/gompi-2023b/extrae-4.2.0/tests/functional/callstack/set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000141348000000000000.mpit
Extrae: Intermediate raw sym file created : /tmp/bot/easybuild/build/Extrae/4.2.0/gompi-2023b/extrae-4.2.0/tests/functional/callstack/set-0/TRACE@aarch64-generic-node2.int.aws-rocky88-202310.eessi.io.0000141348000000000000.sym
Extrae: Deallocating memory.
Extrae: Application has ended. Tracing has been terminated.
FAIL check_Extrae_user_function.sh (exit status: 1)
```

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic

**Comment by ocaisa** [comment]
bot: build repo:eessi.io-2023.06-software arch:x86_64/generic
bot: build repo:eessi.io-2023.06-software arch:aarch64/generic
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/haswell
bot: build repo:eessi.io-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi.io-2023.06-software arch:x86_64/amd/zen4
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi.io-2023.06-software arch:aarch64/neoverse_v1

File: easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.2-2023b.yml [hunk 1]
```diff
@@ -5,3 +5,7 @@ easyconfigs:
       options:
         # see https://github.com/easybuilders/easybuild-easyconfigs/pull/20889
         from-commit: c66c4788a17f7e4f55aa23f9fdb782aad97c9ce7
+  - Extrae-4.2.0-gompi-2023b.eb:
+      options:
+        from-pr: 21017
+        include-easyblocks-from-pr: 3393
```
Comment (bedroge) [inline easystacks/software.eessi.io/2023.06/eessi-2023.06-eb-4.9.2-2023b.yml:11]:
```suggestion
        # see https://github.com/easybuilders/easybuild-easyconfigs/pull/21017
        from-commit: 120f4d56efebd2bc61382db4c84a664a339c66cf
        # see https://github.com/easybuilders/easybuild-easyblocks/pull/3393
        include-easyblocks-from-commit: c4951c78d62fa5cf8e9f6fe0ead212d2a4d7cb9c
```

**Comment by ocaisa** [comment]
@julianmorillo @bedroge Finally!

**Review by bedroge — APPROVED** [review]
Great work!

**Comment by bedroge** [comment]
Hmm, it only uploaded two tarballs... let me try again.

**Comment by bedroge** [comment]
We had run out of disk space on the build cluster. Let's try again...

**Comment by julianmorillo** [comment]
> We had run out of disk space on the build cluster. Let's try again...

This is making us suffer until the end... :-)
